service/src/query_processor.rs

```diff
@@ -209,30 +203,27 @@ mod tests {
 
         let allocation_key =
             attestation_signer_for_allocation(INDEXER_OPERATOR_MNEMONIC, allocation).unwrap();
-        let attestation_signer = create_attestation_signer(
+        let attestation_signer = AttestationSigner::new(
             U256::from(1),
             Address::from_str("0xdeadbeefcafebabedeadbeefcafebabedeadbeef").unwrap(),
             allocation_key,
             *DEPLOYMENT_ID,
-        )
-        .unwrap();
+        );
 
+        let request = "test input";
+        let response = "test output";
         let attestation = QueryProcessor::create_attestation(
             &attestation_signer,
-            "test input".to_string(),
+            request,
             &UnattestedQueryResult {
-                graphql_response: "test output".to_string(),
+                graphql_response: response.to_owned(),
                 attestable: true,
             },
         );
 
-        // Values generated using https://github.com/graphprotocol/indexer/blob/f8786c979a8ed0fae93202e499f5ce25773af473/packages/indexer-native/lib/index.d.ts#L44
-        let expected_signature = Signature {
-            v: 27,
-            r: hex!("a0c83c0785e2223ac1ea1eb9e4ffd4ca867275469a7b73dab24f39ddcdec5466").into(),
-            s: hex!("4d0457efea889f2ec7ffcc7ff9b408428d0691356f34b01f419f7674d0eb4ddf").into(),
-        };
-
-        assert_eq!(attestation, expected_signature);
+        assert_eq!(
+            attestation_signer.verify(&attestation, request, response, &indexer),
+            Ok(())
+        );
```

Comment:
This test fails somehow. Also, I think that'd be a little bit less convoluted:
```suggestion
        attestation_signer.verify(&attestation, request, response, &indexer).unwrap();
```